### ISSUE TITLE
feat(@secretlint/secretlint-rule-github): support new GitHub App installation token format (ghs_)

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1518]
## Background

GitHub announced a new format for GitHub App installation tokens:

- Announcement: https://github.blog/changelog/2026-04-24-notice-about-upcoming-new-format-for-github-app-installation-tokens/
- Rollout: April 27 – June 2026 (GitHub Enterprise Cloud and Data Residency)

### What changes

| | Old format | New format |
|---|---|---|
| Prefix | `ghs_` | `ghs_` (unchanged) |
| Length | ~40 characters | ~520 characters (variable) |
| Structure | `ghs_<36 alphanumeric chars>` | `ghs_<AppID>_<JWT>` |
| Characters | `[A-Za-z0-9_]` | `[A-Za-z0-9_\-.]` (JWT contains dots) |

The JWT portion uses base64url encoding and dot separators (e.g., `header.payload.signature`).

## Problem

The current pattern in `packages/@secretlint/secretlint-rule-github/src/index.ts` (line 103):

```ts
const CLASSIC_GITHUB_TOKEN_PATTERN =
    /(?<!\p{L})(?<type>ghp|gho|ghu|ghs|ghr)_[A-Za-z0-9_]{36}(?![A-Za-z0-9_])/gu;
```

This pattern will **fail to detect** new-format `ghs_` tokens because:

1. `{36}` — expects exactly 36 characters; new tokens are ~520 characters
2. `[A-Za-z0-9_]` — excludes dots (`.`) and hyphens (`-`) present in JWTs
3. `(?![A-Za-z0-9_])` — the negative lookahead would stop matching at the first dot in the JWT, so the 36-char check would also fail

## Expected behavior

`@secretlint/secretlint-rule-github` should detect both old and new `ghs_` token formats.

A possible approach is to add a separate pattern for the new format alongside the existing one:

```ts
// New format: ghs_<numeric AppID>_<JWT> (~520 chars, contains dots and hyphens)
const GITHUB_APP_INSTALLATION_TOKEN_PATTERN =
    /(?<!\p{L})(?<type>ghs)_[0-9]+_[A-Za-z0-9\-_.]{100,}(?![A-Za-z0-9\-_.])/gu;
```

Alternatively, loosen the `ghs` branch of the classic pattern to accommodate the new length and character set.

## References

- https://github.blog/changelog/2026-04-24-notice-about-upcoming-new-format-for-github-app-installation-tokens/
- https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/
